### ISSUE TITLE
Manually applying a door to a wall moves the door

### ISSUE THREAD

**Issue by Gorgious56** [IfcOpenShell/IfcOpenShell#3703]
Maybe related to https://github.com/IfcOpenShell/IfcOpenShell/issues/3620 

I would assume manually applying a door to a wall would at most move the door along its local Y axis (so that its origin is on the wall axis), not along the door local X axis. 

Another thing I would assume is the door would rotate so its local X axis is colinear (or oppositely colinear if that word exists ?) to the wall axis, which it does correctly now.

![Animation3](https://github.com/IfcOpenShell/IfcOpenShell/assets/25156105/03bfd6e9-3199-4e12-a2d7-d17229242562)

**Comment by Gorgious56** [comment]
OK so It seems I was mistaken, it does not move the door along its local X axis, it rotates the door along its origin so the door X and wall X point in the same direction. I'm fine with this behaviour, since we can flip a door once applied to wall, but flipping a door doesn't revolve it around its axis, it revolves it around it center point. I think one of the two behaviours is not right, either flipping a door revolves it around it origin or its center point, should it not ?

![Animation3](https://github.com/IfcOpenShell/IfcOpenShell/assets/25156105/b87eb204-2686-4c95-8f07-e1ac457de609)

**Comment by Moult** [comment]
Yeah there are a number of known flipping bugs that have yet to be resolved.

**Comment by Gorgious56** [comment]
I believe after looking a bit at the docs that in this case the door should be translated along its Y axis so that its X axis is colinear with the interior of the wall. Flipping indeed makes it adhere to the docs but it feels weird.

https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/schema/ifcsharedbldgelements/lexical/ifcdoor.htm
